src/Security/Authorization/Policy/src/IAuthorizationMiddlewareResultHandler.cs

```diff
@@ -0,0 +1,15 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authorization
```

Comment:
It seems to me that Microsoft's .NET libraries tend to have the convention of keeping different types of abstractions (abstract classes, interfaces etc.) in a separate namespace and - sometimes - a separate project/package. 
I find it reasonable; it makes the code well organized and allows to have (more) stable dependencies on these projects (i.e. abstractions NuGet package gets updated rarely and as such it can be a dependency to a custom project that doesn't need to be updated often as an opposite to the implementation parts which get released with i.e. bugfixes or minor releases...

So it it makes me wonder: what's the rule being followed in this regard? Is there any particular reason why this convention is not followed everywhere in aspnetcore (i.e. here)?
We could put the abstractions in a directory which would keep the namespace as `Microsoft.AspNetCore.Authorization.Abstractions` from the very beginning and once it'd be decided that it's worthy to move it into a separate package of its own - there'd be no need to adjust the namespaces...

Reply:
We didn't feel like we needed an abstraction/core split for authorization since it was tiny, we made the same decision for Options for example as well, we skipped having an Abstractions package there as well.  The overhead of extra packages didn't seem worth the benefit

Reply:
I thought that creating an "Abstractions" namespace upfront wouldn't hurt; not much work (just an extra directory) and with the namespace in place you could decide later if it's worth splitting into separate package or not.
It'd be postponing the decision as I see it ;)